nikola/plugins/task/galleries.py

```diff
@@ -229,7 +236,18 @@ def gen_tasks(self):
 
                 image_name_list = [os.path.basename(p) for p in image_list]
 
-                if self.kw['use_filename_as_title']:
+                if self.captions:
```

Comment:
Why is this a global variable? This should be handled in a gallery-specific way. Nikola can be run with threading, in which case breakage will occur.

Instead, return `captions, order` from `find_metadata()` and use that.

Also: how does this work for multilingual sites?

Reply:
Now using  `order, captions = self.find_metadata(gallery)` as suggested, and adding both as elements of `context`.

I haven't got a multilingual site to with, unfortunately. Looking through the code, though, I believe that the only place where non-en might get caught is in the call to json.dumps(). If we specify `ensure_ascii=False` that should do the trick.